faucet/components/Faucet.tsx

```diff
@@ -0,0 +1,121 @@
+import { useState, useEffect } from 'react'
+import { isAddress } from 'web3-utils'
+import type { Networks } from '@hoprnet/hopr-ethereum/scripts/addresses'
+
+async function mint(
+  network: Networks,
+  address: string
+): Promise<{
+  success: boolean
+  transactionHash?: string
+  message?: string
+}> {
+  const response = await fetch(`/api/mint?network=${network}&address=${address}`)
+
+  if (response.status === 200) {
+    const { transactionHash } = await response.json()
+    return {
+      success: true,
+      transactionHash,
+    }
+  } else {
+    return {
+      success: false,
+      message: response.statusText,
+    }
+  }
+}
+
+function Faucet({ network }: { network: Networks }) {
+  const [address, setAddress] = useState<string>(undefined)
+  const [isValidAddress, setIsValidAddress] = useState<boolean>(undefined)
+  const [status, setStatus] = useState<'PENDING' | 'SUCCESS' | 'FAILURE'>(undefined)
+  const [message, setMessage] = useState<string>(undefined)
+  const isButtonDisabled = !isValidAddress || status === 'PENDING'
+  const showStatusMessage = status === 'SUCCESS' || status === 'FAILURE'
+
+  // update 'isValidAddress' on 'address' change
+  useEffect(() => {
+    if (typeof address === 'undefined' || address === '') {
+      setIsValidAddress(undefined)
+    } else {
+      setIsValidAddress(isAddress(address))
+    }
+  }, [address])
```

Comment:
```suggestion
  useEffect(() => {
    !!address ? setIsValidAddress(isAddress(address) : setIsValidAddress(undefined)
  }, [address])
```

Reply:
I like to be as explicit as possible, for me it takes a bit more brain processing power to understand what's going on by looking at the suggestion, also I should add a few comments here.

This seems like coding preference so perhaps we can discuss this with @robertkiel and figure some guidelines / best practices?